internal/graph/lookup.go

```diff
@@ -260,10 +262,17 @@ func (cl *ConcurrentLookup) lookupDirect(ctx context.Context, req ValidatedLooku
 			// For each inferred object found, check for the target ONR.
 			objects := tuple.NewONRSet()
 			if len(result.Resp.ResolvedOnrs) > 0 {
-				it, err := cl.ds.QueryTuples(datastore.TupleQueryResourceFilter{
-					ResourceType:             req.ObjectRelation.Namespace,
-					OptionalResourceRelation: req.ObjectRelation.Relation,
-				}, req.Revision).WithUsersets(result.Resp.ResolvedOnrs).Limit(uint64(req.Limit)).Execute(ctx)
+				limit := uint64(req.Limit)
+				it, err := cl.ds.QueryTuples(
+					ctx,
+					&v1_proto.RelationshipFilter{
+						ResourceType:     req.ObjectRelation.Namespace,
+						OptionalRelation: req.ObjectRelation.Relation,
+					},
+					req.Revision,
+					options.SetUsersets(result.Resp.ResolvedOnrs),
+					options.WithLimit(&limit),
```

Comment:
`WithLimit` taking in a pointer makes sense to define no limit, but the requirement to `&theIntValue` isn't great... could we have a `WithLimit(uint64)` and `WithNoLimit`, or would that break the optgen?

Reply:
optgen is not magic, it sees a type and it builds a `WithX` function using the same type. @ecordell correct me if I'm wrong.